dsa-solutions/gfg-solutions/0001-value-equal-to-index-value.md

```diff
@@ -0,0 +1,124 @@
+
+## Constraints
+
+* 1 ≤ N ≤ 105
+* 1 ≤ `Arr[i]` ≤ 106
+
+## Problem Explanation
+
+The problem involves finding elements in an array whose values match their index positions, considering 1-based indexing. This means if an element's value matches its position in the array when counting from 1 (not 0), it should be included in the result.<br />
+
+Example:
+
+* Given an array [15, 2, 45, 12, 7]:
+    - At 0-based index 0, the value is 15 (not equal to 1)
+    - At 0-based index 1, the value is 2 (equal to 2)
+    - At 0-based index 2, the value is 45 (not equal to 3)
+    - At 0-based index 3, the value is 12 (not equal to 4)
+    - At 0-based index 4, the value is 7 (not equal to 5)
+
+The only element that matches its index is 2 at index 2.
+
+## Python Solution Code
+
+```py
+class Solution:
+    
+    def valueEqualToIndex(self, arr, n):
+        result = []
+        for i in range(n):
+            if (arr[i] == i + 1):
+                result.append(arr[i]);
+        return result
+```
+
+## C++ Solution Code
+
+```cpp
+class Solution {
+    public:
+    	vector<int> valueEqualToIndex(int arr[], int n) {
+    	    vector<int> result;
+    	    for(int i = 0; i < n; i++) {
+    	        if(arr[i] == i + 1)
+    	            result.push_back(arr[i]);
+    	    } return result;
+        }
+};
+```
+
+## Solution Logic
+
+1. Initialize an empty result list: This will store the elements that satisfy the condition.
+2. Iterate through the array: Use a loop to go through each element in the array.
+3. Check the condition: For each element, check if the element's value matches its 1-based index. Since array indexing typically starts at 0 in most programming languages, you adjust by adding 1 to the index.
+4. Store matching elements: If an element's value matches its 1-based index, add it to the result list.
+5. Return the result list: After checking all elements, return the list of elements that met the condition.
+
+## Example Walkthrough
+
+For the array `[15, 2, 45, 12, 7]` with length `n = 5`:
+
+* Initialize an empty list: `result = []`
+* Loop through the array with indices from 0 to 4:
+    - Check element at index 0 (1-based index 1): 15 ≠ 1
+    Check element at index 1 (1-based index 2): 2 = 2 (add to result)
+    Check element at index 2 (1-based index 3): 45 ≠ 3
+    Check element at index 3 (1-based index 4): 12 ≠ 4
+    Check element at index 4 (1-based index 5): 7 ≠ 5
+
+Final `result` list: `[2]`
+
+## Time Complexity
+
+The time complexity of this solution is O(N) where N is the number of elements in the array. This is because the solution involves a single pass through the array to check each element against its 1-based index.
+
+## Space Complexity
+
+The space complexity is O(1) for the input space, and O(K) for the output space, where K is the number of elements matching the condition. In the worst case, if all elements match, the space used by the result list will be O(N).
```

Comment:
basic practice:

```md
O(1)
```

Good Practice:

```md
$O(1)$
```

and 

basic practice:

```md
O(N)
```

Good Practice:

```md
$O(n)$
```